### ISSUE TITLE
It doesn't work aws-amplify

### ISSUE THREAD

**Issue by nishidashib** [fuse-box/fuse-box#1410]
It doesn't work aws-amplify.
i added aws-amplify

```
$ yarn add --dev aws-amplify.
$ rm -rf .fusebox/
$ rm -rf .node_modules/
$ yarn start
```
my test repo: https://github.com/nishidashib/amplify-not-working
but an error occurred after building
<img width="1191" alt="2018-10-14 4 10 37" src="https://user-images.githubusercontent.com/3824491/46909054-2be7e900-cf67-11e8-9336-123056c6bce7.png">

```
fuse-box: 3.5.0
yarn: 1.10.1
node: 10.11.0  or 8.4.0

vue.js 2.5.16
typescrpt 3.1.12
```

if i comment out the follow in index.vendor.js
```
:line 19255 
// util.domain = require('domain');

line 28687 
// var dgram = require('dgram');  

line 75166-75170
//   SAXStream.prototype = Object.create(Stream.prototype, {
//     constructor: {
//       value: SAXStream
//     }
//   })
```
it looks like aws-amplify works

<img width="1272" alt="2018-10-14 6 54 37" src="https://user-images.githubusercontent.com/3824491/46910396-15995780-cf7e-11e8-9143-71349db046d0.png">

vue-cli with webpack works, but i want to use fuse-box.
what is wrong??

Please tell me the cause...

**Comment by nchanged** [comment]
It looks like we are missing domain package polyfills
Can you try installing it?

**Comment by nishidashib** [comment]
what is domain package polyfills?'
i don't know.

i tried follows again

The result does not change
```
$ rm -rf .fusebox/
$ rm -rf .node_modules/
$ yarn build
```
domain package is used by aws-sdk-js/lib/node_loader.js(aws-amplify dependency)

build was failed  at  require('domain') and require('dgram') 
```
・・・
util.crypto.lib = require('crypto');
util.Buffer = require('buffer').Buffer;

util.domain = require('domain'); // build was failed  with fuse-box. require('dgram') too
util.stream = require('stream');
・・・
```

**Comment by nchanged** [comment]
Yarn add domain

That's what I meant

**Comment by nishidashib** [comment]
of course i did 

```
$ yarn add --dev domain
```

but 
```
index.vendor.js:60983 Uncaught ReferenceError: System is not defined
    at Object.<anonymous> (index.vendor.js:60983)
    at c (index.vendor.js:120782)
    at b.require (index.vendor.js:120782)
    at Object.<anonymous> (index.vendor.js:60978)
    at c (index.vendor.js:120782)
    at b.require (index.vendor.js:120782)
    at Object.<anonymous> (index.vendor.js:25224)
    at c (index.vendor.js:120782)
    at b.require (index.vendor.js:120782)
```
<img width="789" alt="2018-10-17 17 57 55" src="https://user-images.githubusercontent.com/3824491/47074851-3c81b300-d236-11e8-9816-477343d28427.png">

systemjs??

**Comment by nchanged** [comment]
Let's try the following
1) create [modules folder](https://fuse-box.org/docs/development/configuration#custom-modules-folder)
2) Add `domain` folder
3) Create index.js with this [content](https://github.com/bevry/domain-browser/blob/master/source/index.js)

If it will work I will make a polyfill or you could help out add it yourself [here](https://github.com/fuse-box/fuse-box/tree/master/modules) in a similar fashion. 

We are just missing the polyfill

**Comment by nishidashib** [comment]
ok.

but an error occurred after building
```
index.vendor.js:93837 Uncaught Package not found dgram
```
 i can' found content which i can create index.js.
please let me know.

**Comment by nchanged** [comment]
I guess we are looking at this package too https://github.com/icopp/browser-shim-node-dgram

**Comment by nchanged** [comment]
@nishidashib can you please reproduce the issue? I will push the missing libs and release a next version

**Comment by nchanged** [comment]
@nishidashib it has been fixed

**Comment by nishidashib** [comment]
@nchanged 

sorry , my response was late.
but  the error was happend.

fuse-box 3.6.0
```
Acorn error: 'return' outside of function (67:1)
File: /Users/takeshishibasaki/Documents/data-section/docker-front/console/node_modules/fuse-box/modules/domain/index.js

             return d
     }
      return domain
  }
```

**Comment by nishidashib** [comment]
@nchanged 

the error was happend too (fuse-box 3.6.0)
```
index.vendor.js:75620 Uncaught TypeError: Cannot read property 'prototype' of undefined
    at index.vendor.js:75620
    at Object.<anonymous> (index.vendor.js:76963)
    at c (index.vendor.js:94166)
    at b.require (index.vendor.js:94166)
    at Object.<anonymous> (index.vendor.js:90511)
    at Object.<anonymous> (index.vendor.js:90859)
    at c (index.vendor.js:94166)
    at b.require (index.vendor.js:94166)
    at Object.<anonymous> (index.vendor.js:90266)
    at Object.<anonymous> (index.vendor.js:90291)
```

```
  SAXStream.prototype = Object.create(Stream.prototype, {
    constructor: {
      value: SAXStream
    }
  })
```

my test repo: https://github.com/nishidashib/amplify-not-working

**Comment by riglar** [comment]
I patched this as follows:
- go to node_modules/fuse-box/modules/
- to fix domain polyfill - edit domain/index.js and comment out line 67 `return domain`
- to fix steam polyfill - `yarn add --dev stream-browserify`
- edit stream/index.js and change line 4 to `module.exports = { Stream: require("stream-browserify") };`

@nchanged - would you like a PR for this?

**Comment by nchanged** [comment]
@riglar please do that ;-) thanks

**Comment by riglar** [comment]
Should be fixed by #1442
